plugins/pipenv/pipenv.plugin.zsh

```diff
@@ -0,0 +1,51 @@
+#compdef pipenv
+_pipenv() {
+  eval $(env COMMANDLINE="${words[1,$CURRENT]}" _PIPENV_COMPLETE=complete-zsh  pipenv)
+}
+
+_togglePipenvShell() {
+  # deactivate shell if Pipfile doesn't exit and not in a sub dir
+  if [[ ! -a "$PWD/Pipfile" ]]; then
+      if [[ "$PIPENV_ACTIVE" == 1 ]]; then
+          if [[ "$PWD" != "$pipfile_dir"* ]]; then
+            exit
+          fi
+    fi
+  fi
+
+  # activate the shell if Pipfile exists
+  if [[ "$PIPENV_ACTIVE" != 1 ]]; then
+    if [[ -a "$PWD/Pipfile" ]]; then
+      export pipfile_dir="$PWD"
+      pipenv shell
+    fi
+  fi
+}
+
+if [[ "$(basename ${(%):-%x})" != "_pipenv" ]]; then
+  autoload -U compinit && compinit
```

Comment:
This is already done by OMZ, you don't need to call compinit again. Also, the check for file name above (`if [[ $(basename ...`) is superfluous because it will always return `pipenv.plugin.zsh`, hence it will always execute.

I would just leave the `compdef _pipenv pipenv` and put it right below the `_pipenv` function.